controllers/manilaapi_controller.go

```diff
@@ -638,30 +638,43 @@ func (r *ManilaAPIReconciler) reconcileNormal(ctx context.Context, instance *man
 	//
 	// check for required OpenStack secret holding passwords for service/admin user and add hash to the vars map
 	//
-	ctrlResult, err := r.getSecret(ctx, helper, instance, instance.Spec.TransportURLSecret, &configVars)
+
+	hash, ctrlResult, err := ensureSecret(
+		ctx,
+		types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.TransportURLSecret},
+		[]string{
+			"transport_url",
+		},
+		helper.GetClient(),
+		&instance.Status.Conditions,
+		manila.NormalDuration,
+	)
 	if err != nil {
 		return ctrlResult, err
 	}
```

Comment:
Same question here about checking for `ctrlResult != ctrl.Result{}`

Reply:
same answer applies here

Reply:
Done